empyrical/stats.py

```diff
@@ -702,16 +906,49 @@ def beta(returns, factor_returns, risk_free=0.0):
     float
         Beta.
     """
+    if len(returns) < 2 or len(factor_returns) < 2:
+        return np.nan
+
+    return beta_aligned(*_aligned_series(returns, factor_returns),
+                        risk_free=risk_free)
+
+
+def beta_aligned(returns, factor_returns, risk_free=0.0):
+    """Calculates beta.
+
+    If they are pd.Series, expects returns and factor_returns have already
+    been aligned on their labels.  If np.ndarray, these arguments should have
+    the same shape.
+
+    Parameters
+    ----------
+    returns : pd.Series or np.ndarray
+        Daily returns of the strategy, noncumulative.
+        - See full explanation in :func:`~empyrical.stats.cum_returns`.
+    factor_returns : pd.Series or np.ndarray
+         Daily noncumulative returns of the factor to which beta is
+         computed. Usually a benchmark such as the market.
+         - This is in the same style as returns.
+    risk_free : int, float, optional
+        Constant risk-free return throughout the period. For example, the
+        interest rate on a three month us treasury bill.
+
+    Returns
+    -------
+    float
+        Beta.
+    """
 
     if len(returns) < 2 or len(factor_returns) < 2:
         return np.nan
     # Filter out dates with np.nan as a return value
-    joint = pd.concat([_adjust_returns(returns, risk_free),
-                       factor_returns], axis=1).dropna()
-    if len(joint) < 2:
+    joint = np.vstack([_adjust_returns(returns, risk_free),
```

Comment:
Ah, but this is in `beta_aligned` now, so we expect them to be the same size already.  zipline's arrays are already aligned, but if a consumer can't guarantee that, they should use the existing [`beta` function] (https://github.com/quantopian/empyrical/pull/32/files/4d24d3f551e76b6027ceb804b4d1d5362d1e5b68#diff-23342a2bf4f4c31bfe2dfe0bcbe54c8bR912), which calls this after aligning them.  `beta` is what pyfolio still calls.

Does that sound ok to you?